src/Sfa.Tl.Matching.Web/Controllers/ProviderController.cs

```diff
@@ -117,9 +125,43 @@ public async Task<IActionResult> ConfirmProviderChange(HideProviderViewModel vie
                 return View("ConfirmProviderChange", viewModel);
             }
 
-            await _providerService.SetIsProviderEnabledForSearchAsync(viewModel.ProviderId, !viewModel.IsEnabledForSearch);
+            viewModel.IsEnabledForSearch = !viewModel.IsEnabledForSearch;
+            await _providerService.UpdateProviderAsync(viewModel);
 
             return RedirectToRoute("GetProviderDetail", new { providerId = viewModel.ProviderId });
         }
+
+        [HttpPost]
+        [RequestFormLimits(ValueCountLimit = 5000)]
+        [Route("save-provider-feedback", Name = "SaveProviderFeedback")]
+        public async Task<IActionResult> SaveProviderFeedback(SaveProviderFeedbackViewModel viewModel)
+        {
+            await _providerService.UpdateProvider(viewModel);
```

Comment:
There was a requirement to save first before sending the email so that users could tick the check box and send directly to the providers. This is being removed next sprint anyway, so I'll leave for now.